plunger/plunger.py

```diff
@@ -238,6 +199,20 @@ def get_pending_transactions_from_parity(self) -> list:
         items = filter(lambda item: item['blockNumber'] is None, items)
         return list(map(lambda item: Transaction(tx_hash=item['hash'], nonce=int(item['nonce'], 16)), items))
 
+    def get_pending_transactions_from_block(self) -> list:
+        # Get the list of pending transactions from the mempool
+        # First, execute the RPC call and get the response
+        request = {"method": "eth_getBlockByNumber", "params": ["pending", True], "id": 1, "jsonrpc": "2.0"}
+        response = requests.post(self.web3.provider.endpoint_uri + "/rpc", None, request).json()
+
+        # Then extract pending transactions sent by us from the response and convert them into `Transaction` objects
+        items = response['result']['transactions']
+        items = filter(lambda item: item['from'].lower() == self.web3.eth.defaultAccount.lower(), items)
+        result = []
+        for item in list(items):
+            result.append(Transaction(tx_hash=item['hash'], nonce=int(item['nonce'], 16)))
```

Comment:
What is the purpose of the 16?

Reply:
JSON-RPC reports the nonce as a hex string for some strange reason; the `16` converts it to a decimal integer.

Reply:
Ahh, right. That makes sense